doc/versioning.md

```diff
@@ -0,0 +1,19 @@
+# Versioning
+
+SkyPortal being an API-based service, its version number should have a meaning for the end-user AND the developer or system administrator that will deploy it. On the other hand, maintaining 2 distinct version numbers would not be practical.
+
+The development cycle of SkyPortal is such that breaking API changes are rare, as we often add, improve, refactor, or fix existing features rather than change their output or behavior. Version changes of critical system dependencies are also very rare (e.g. upgrading the version of PostgreSQL or the Python runtime). On the other hand, new features (new endpoints, new tables, additional data returned by existing endpoints) are added more frequently. Lastly, changes that do not impact the user nor the developer (e.g. refactoring, performance improvements, bug fixes) are very common, but do not necessarily need to be advertised to the end-user.
+
+Based on these considerations, we have decided to use [Semantic Versioning](https://semver.org/) for SkyPortal. The version number will be of the form `MAJOR.MINOR.PATCH`, where:
+
+- `MAJOR` is incremented when breaking changes are introduced, for the user or the developer
+- `MINOR` is incremented when new features are added in a backwards-compatible manner, requiring no changes to the user's code and minimal changes to the deployment
+- `PATCH` is incremented when backwards-compatible bug fixes are introduced, or when changes are made that do not impact the user nor the developer
+
+The development of SkyPortal is active enough that it would be impractical to update the version number for every commit. Instead, we will update the version number along with a release on GitHub that tags a specific commit in the repository, at most one a week. That way, the developer / system admin can easily checkout to the latest tagged release's commit (when there is a new version available) rather than simply point to whatever is on main. The end-user can always easily check which version of SkyPortal they are querying/using, as it is returned in every API response and displayed on the about page of the web application.
```

Comment:
I reviewed up to here; can I capture what was discussed on the call this morning for this sentence? I.e., we have, say, ~monthly release cadences, where new features get introduced early in the cycle, then tested, and a release is made once everything's stabilized?

Reply:
Yep! I need to edit that based on the call indeed

Reply:
Edited, please let me know what you think!